src/components/UpdateDB/index.js

```diff
@@ -48,31 +58,67 @@ const UpdateDb = () => {
       failedUpload: false,
       successUpload: null,
     });
+    
+    const chunkedIPC = chunk(upload.ipcData, 1000);
+    const fetchAChunk = (chunk) => {
+      return fetch(`${process.env.REACT_APP_API}/donations/upload`, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+          Authorization: "Bearer " + localStorage.getItem("token"),
+        },
 
-    const validateUpSert = (res) => {
-      if (res.ok) {
-        res.json().then((data) => success(data));
-      } else {
-        return failed();
-      }
+        body: JSON.stringify(chunk),
+      })
+        .then((res) => {
+          if (res.ok) return res.json();
+          else throw new Error("Error in uploading chunk");
+        })
+        .then((data) => {
```

Comment:
This `.then(...)` is not necessary since you are just returning back the same data from the previous then-block.